Use CFBundleLocalizations for iOS localization

ThijmenDeValk opened the pull request:
Simplifies the iOS implementation by using CFBundleLocalizations in the Info.plist instead of the convoluted method of creating empty placeholder files.

Fixes #20 and closes #23

ThijmenDeValk committed:
feat: simplify iOS implementation using CFBundleLocalizations (#23 and #20)

bitcrumb commented on plugin/src/withIosLocalizableProject.ts:
```diff
@@ -1,34 +1,16 @@
+    config.modResults["LOCALES_SUPPORTED"] = locales.join(",");
+    config.modResults["CFBundleAllowMixedLocalizations"] = true;
+    config.modResults["CFBundleLocalizations"] = locales;
+    config.modResults["CFBundleDevelopmentRegion"] = "en";
```
Just a small heads up that you might want to make this configurable (and maybe the `CFBundleAllowMixedLocalizations` as well).

ThijmenDeValk replied:
Good point. I do think these are the sane defaults for folks that don't explicitly set them, though.

bitcrumb replied:
Agree, no harm in having a sane default. You could surely have the config optional and fallback to these as default :)

I am testing out the beta as we speak btw.

ThijmenDeValk committed:
fix: make example project build again
fix: don't link to specific Expo version
fix: run expo-module prepare in prepare script
chore: update deps

bitcrumb commented:
Any chance we can get this released? 😁

ThijmenDeValk commented:
@bitcrumb Sorry, been a busy couple of weeks 😅 I'll get it out the door later this week. Had some tweaks I wanted to make for Expo 52 first.
